### ISSUE TITLE
Bug: Contracts are reported as not found

### ISSUE THREAD

**Issue by amanusk** [eqlabs/pathfinder#1135]
Running a local pathfinder (0.5.6) goerli node.
Pathfinder run cmd:
```
./target/release/pathfinder --network testnet --ethereum.url https://eth-goerli.g.alchemy.com/v2/<INFURA> --data-directory /media/ultra/chains/pathfinder --http-rpc 192.168.1.45:5235 --poll-pending true
```

Running the following snippet:
```
import { RpcProvider } from "starknet";

const STARKNET_RPC_URL = "http://192.168.1.45:5235";
// const STARKNET_RPC_URL = "https://starknet-goerli.infura.io/v3/INFURA";

const provider = new RpcProvider({ nodeUrl: STARKNET_RPC_URL });

async function main() {
  let block: any = await provider.getBlockNumber();
  console.log(block);

  let classHash: any = await provider.getClassHashAt(
    "0x03e85bfbb8e2a42b7bead9e88e9a1b19dbccf661471061807292120462396ec9",
  );
  console.log("class hash", classHash);
}

main();
```

Getting the latest block, but contract not found:
```
818732
/home/amanusk/Code/Cairo0/queries/node_modules/starknet/src/provider/rpc.ts:85
      throw new LibraryError(`${code}: ${message}`);
            ^
LibraryError: 20: Contract not found
    at RpcProvider.errorHandler (/home/amanusk/Code/Cairo0/queries/node_modules/starknet/src/provider/rpc.ts:85:13)
    at RpcProvider.fetchEndpoint (/home/amanusk/Code/Cairo0/queries/node_modules/starknet/src/provider/rpc.ts:96:12)
```

same snippet with infura:
```
818732
class hash 0x1cb96b938da26c060d5fd807eef8b580c49490926393a5eeb408a89f84b9b46
```

**Comment by Mirko-von-Leipzig** [comment]
Hmm. Curious, I'm unable to replicate this issue -- and infura also runs pathfinder so it seems to be just your local pathfinder node that's misbehaving.

How long have you been running the pathfinder node? Any idea what version you started with?

**Comment by amanusk** [comment]
Running for quite a while. Definitely since ~0.4.X versions.
Only noticed this recently. The same source code is used for mainnet, which runs fine.
Thought might use the op to avoid a full re-sync and/or detect an issue :)

**Comment by Mirko-von-Leipzig** [comment]
A couple of potential reasons this might occur:
1. we had a bug at some stage during which you sync'd some blocks which are impacted this query
2. we had a buggy migration
3. your disk is malfunctioning or had a bit flip or something

Could you try the following to poke around the database a bit:

1. stop pathfinder
2. open the database using sqlite3 - you may need to install sqlite3 first.
```
sqlite3 /media/ultra/chains/pathfinder/goerli.sqlite
```
You should see something like
```
SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for usage hints.
sqlite>
```
Then we can start poking around. 

Let's see if the database contains the class in question:
```
SELECT COUNT(1) FROM class_definitions WHERE hash = x'01cb96b938da26c060d5fd807eef8b580c49490926393a5eeb408a89f84b9b46';
```
This should output `1`.

And let's see the contract's class hash, and the block number it was deployed at:
```
SELECT block_number, hex(class_hash) FROM contract_updates WHERE contract_address = x'03e85bfbb8e2a42b7bead9e88e9a1b19dbccf661471061807292120462396ec9';
```
For me this outputs:
```
130593|01CB96B938DA26C060D5FD807EEF8B580C49490926393A5EEB408A89F84B9B46
```
i.e.
```
block number: 130 593
class hash:   01CB96B938DA26C060D5FD807EEF8B580C49490926393A5EEB408A89F84B9B46
```

**Comment by amanusk** [comment]
```
SELECT COUNT(1) FROM class_definitions WHERE hash = x'01cb96b938da26c060d5fd807eef8b580c49490926393a5eeb408a89f84b9b46';
```

Returns 1

```
sqlite> SELECT block_number, hex(class_hash) FROM contract_updates WHERE contract_address = x'03e85bfbb8e2a42b7bead9e88e9a1b19dbccf661471061807292120462396ec9';
```

Returns blank

**Comment by Mirko-von-Leipzig** [comment]
Weird. Let's see if there is anything in that table..
```
SELECT COUNT(1) FROM contract_updates;
SELECT block_number FROM contract_updates ORDER BY block_number DESC LIMIT 10;
```

**Comment by Mirko-von-Leipzig** [comment]
I believe we actually fixed this in #1318 and released in [v0.7.2](https://github.com/eqlabs/pathfinder/releases/tag/v0.7.2). If this is still occurring please re-open.
